### ISSUE TITLE
Closing orphaned open sessions

### ISSUE THREAD

**Issue by always-developing** [OpenSC/OpenSC#3176]
### Problem Description

[Question] - is it possible to close all sessions (specifically orphaned sessions) to a slot?

I am executing the following command to re-initialize a slot (which was previously successfully initialized, but I now want to re-initialize for testing purposes)

`pkcs11-tool --module /usr/lib/libckteec.so.0 --init-token --slot 2 --label blankslot --so-pin secretpin`

However the above is returning the following:

`error: PKCS11 function C_InitToken failed: rv = CKR_SESSION_EXISTS (0xb6)`

My application which uses the slot is not running at the time of executing the command. I have also restarted the machine - the return message persists.

Basically, is appears as through a session has been opened and not closed, and I cannot track down which process has the open session.

Is there anyway to see open connections to a slot and close them if required?

Thanks.

**Comment by Jakuje** [comment]
There is `C_CloseAllSessions()` which could solve the issue for you. Generally on pkcs11 layer, the sessions are local to the process or to the driver -- open sessions should not be kept on the HW device, certainly not over power-off cycle.

This error sounds like there is something else (or even the pkcs11-tool) connected to the pkcs11 module. This sounds like the pkcs11 module is using some client-server architecture so in that case I would suggest to check with the manufacturer/module developer how the initialization should be handled or if there is a way to achieve this.

**Comment by always-developing** [comment]
Thanks for the feedback @Jakuje  - even after a proper power-off cycle, the problem persisted. This lead us to believe that there was a process holding a session - there was application running, which was accessing a **different slot**, this was however preventing the re-initializing of my slot. Stopping the process accessing the other slot solved the issue.
